src/ophyd_async/core/_table.py

```diff
@@ -17,34 +22,31 @@ def _concat(value1, value2):
 class Table(BaseModel):
     """An abstraction of a Table of str to numpy array."""
 
-    model_config = ConfigDict(validate_assignment=True, strict=False)
-
-    @staticmethod
-    def row(cls: type[TableSubclass], **kwargs) -> TableSubclass:  # type: ignore
-        arrayified_kwargs = {}
-        for field_name, field_value in cls.model_fields.items():
-            value = kwargs.pop(field_name)
-            if field_value.default_factory is None:
-                raise ValueError(
-                    "`Table` models should have default factories for their "
-                    "mutable empty columns."
-                )
-            default_array = field_value.default_factory()
-            if isinstance(default_array, np.ndarray):
-                arrayified_kwargs[field_name] = np.array(
-                    [value], dtype=default_array.dtype
-                )
-            elif issubclass(type(value), Enum) and isinstance(value, str):
-                arrayified_kwargs[field_name] = [value]
+    # Allow extra so we can use this as a generic Table
+    model_config = ConfigDict(extra="allow")
```

Comment:
I'm not sure how this works, given the different config in `__init_subclass__`, this config only applies to fields of the `Table` base class, and there aren't any...?

Reply:
You can use `Table` in 2 ways:
1. `Table(**whatever_pva_gives_us)` when pvi adds a Signal to a Device that is not type hinted
2. `MyTable(**whatever_pva_gives_us)` where the Signal is type hinted

For 1 we want extra="allow" so it is passed through as is. There are no base class fields, only "extra" fields, so they must be allowed. For 2 we want extra="forbid" so it is strictly checked against the BaseModel we are supplied.